test/test_nn.py

```diff
@@ -14315,6 +14318,67 @@ def test_MaxUnpool_zero_batch_dim(self, device):
         self.assertEqual(inp.grad, torch.zeros_like(inp))
         self.assertEqual(unpool_out, torch.zeros_like(unpool_out))
 
+    @slowTest
+    @onlyNativeDeviceTypes
+    def test_MaxUnpool_index_errors(self, device):
+        test_cases = [
+            # module_name, module_size, output_size, test_index, should_error
+            ('MaxUnpool2d', (2, 2), (1, 3, 4, 5), -1, True),
+            ('MaxUnpool2d', (2, 2), (1, 3, 4, 5), 2 * 2 * 4 * 5, True),
+            ('MaxUnpool2d', (2, 2), (1, 3, 4, 5), (2 * 2 * 4 * 5) - 1, False),
+            ('MaxUnpool2d', (2, 3), (2, 1, 4, 2), 2 * 3 * 4 * 2, True),
+            ('MaxUnpool2d', (2, 3), (2, 1, 4, 2), (2 * 3 * 4 * 2) - 1, False),
+
+            ('MaxUnpool3d', (2, 2, 2), (1, 3, 4, 5), -1, True),
+            ('MaxUnpool3d', (2, 2, 2), (1, 3, 4, 5), 2 * 2 * 2 * 3 * 4 * 5, True),
+            ('MaxUnpool3d', (2, 2, 2), (1, 3, 4, 5), (2 * 2 * 2 * 3 * 4 * 5) - 1, False),
+            ('MaxUnpool3d', (2, 2, 2), (2, 3, 4, 1), 2 * 2 * 2 * 3 * 4 * 1, True),
+            ('MaxUnpool3d', (2, 2, 2), (2, 3, 4, 1), (2 * 2 * 2 * 3 * 4 * 1) - 1, False),
+        ]
```

Comment:
FYI you can now use the `@parametrize_test` decorator to handle things like this. It could look something like this here:

```python
@slowTest
@onlyNativeDeviceTypes
@parametrize_test("module_name,module_size,output_size,test_index,should_error", [
    subtest(('MaxUnpool2d', (2, 2), (1, 3, 4, 5), -1, True), name='optional_subtest_name'),
    # other subtests here
    ...
])
def test_MaxUnpool_index_errors(self, device, module_name, module_size, output_size, test_index, should_error):
    ...
```

This will generate a test per subtest that can be run in isolation if needed.

Reply:
Done, that's a cool new feature